hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/protocol/ClientProtocol.java

```diff
@@ -481,9 +484,6 @@ void cancelDelegationToken(Token<OzoneTokenIdentifier> token)
    */
   S3SecretValue getS3Secret(String kerberosID) throws IOException;
 
-  @VisibleForTesting
```

Comment:
I have a question, we are removing this public utility method from client protocol, this will break applications if they use this method right, after the upgrade. Or end users(clients) should not use the RpcClient class directly only they should access through FileSystem and S3 Interfaces is the contract?

Reply:
End users shouldn't use any public method which are annotated with `@VisibleForTesting` or `@Private`. And they should be careful if `@Stable` is missing from the class or methods. So I think it's not a problem here.

But I agree. And this is the reason why I think this method (independent from the patch) should be added only for the implementation not for the interface. If it's `VisibleForTesting`, we can do an easy cast during the unit test.